src/main/java/view/PostView.java

```diff
@@ -88,13 +107,66 @@ public PostView(PostController postController, PostViewModel postViewModel,
                 0,
                 new Font(FONT_TYPE, Font.BOLD, 14),
                 Color.DARK_GRAY));
-            
+
         mainContent.add(titlePanel, BorderLayout.NORTH);
         mainContent.add(new JScrollPane(postContent), BorderLayout.CENTER);
         mainContent.add(new JScrollPane(commentsPanel), BorderLayout.SOUTH);
         add(mainContent);
+
+        // the button for options
+        JButton optionsButton = new JButton("⋮");
+        JPopupMenu optionsMenu = new JPopupMenu();
+        JMenuItem deletePostItem = new JMenuItem("Delete Post");
+
+        optionsMenu.add(deletePostItem);
+
+        int buttonWidth = 30;
+        int buttonHeight = 30;
+        optionsButton.setBounds(POST_WIDTH - buttonWidth - 10, 10, buttonWidth, buttonHeight);
+        titlePanel.add(optionsButton);
+
+        optionsButton.addActionListener(e -> optionsMenu.show(optionsButton, 0, optionsButton.getHeight()));
+
+        deletePostItem.addActionListener(e -> {
+            int result = JOptionPane.showConfirmDialog(
+                    optionsButton,
+                    "Are you sure you want to delete?",
+                    "Confirm Delete",
+                    JOptionPane.YES_NO_OPTION,
+                    JOptionPane.WARNING_MESSAGE
+            );
+
+            if (result == JOptionPane.YES_OPTION) {
+                String currentUserId = userRepo.getCurrentUser().getUserID();
```

Comment:
I think this violates clean architecture for MVC design, you shouldn't access the current user through the dao, but rather the GetUserInteractor - and get the entire user object instead of just the id, so u can pass it in ur input data